secrets/hashicorpvault/hashicorp_vault.go

```diff
@@ -91,17 +85,48 @@ func (v *VaultSecretsManager) Setup() error {
 		return fmt.Errorf("unable to initialize Vault client: %v", err)
 	}
 
-	// Set the access token
-	client.SetToken(v.token)
-
-	// Set the namespace 
+	// Set the namespace
 	client.SetNamespace(v.namespace)
 
 	v.client = client
 
+	token, err := v.login()
+	if err != nil {
+		return fmt.Errorf("login authentication error: %v", err)
+	}
+
+	// Set the access token
+	client.SetToken(token.Auth.ClientToken)
+
 	return nil
 }
 
+// AWS ec2 auth type
+func (v *VaultSecretsManager) login() (*vault.Secret, error) {
+
+	awsAuth, err := auth.NewAWSAuth(
+		auth.WithEC2Auth(),
+		auth.WithRole("foundation-node-role"),
+		// override dynamic nonce generation
+		auth.WithNonce("static-nonce"),
```

Comment:
Here I used a non-changing nonce since generating a new nonce and saving to file each time doesn't offer any added security. 

This can also be passed into the secrets config..